src/protocolsupportpocketstuff/hacks/playerheads/PlayerHeadsPacketListener.java

```diff
@@ -0,0 +1,617 @@
+				cachedSkullBlocks.get(asLong).destroy(this);
+				cachedSkullBlocks.remove(asLong);
+			}
+			return;
+		}
+		if (packetId == PEPacketIDs.CHUNK_DATA) {
+			int chunkX = VarNumberSerializer.readSVarInt(data); // chunk X
+			int chunkZ = VarNumberSerializer.readSVarInt(data); // chunk Z
+			VarNumberSerializer.readVarInt(data); // length
+			int sectionLength = data.readByte();
+
+			int chunkXStart = chunkX << 4;
+			int chunkZStart = chunkZ << 4;
+
+			HashMap<Long, Integer> skulls = new HashMap<>();
+
+			for (int idx = 0; sectionLength > idx; idx++) {
+				data.readByte(); // storage type
+				for (int x = 0; x < 16; x++) {
+					for (int z = 0; z < 16; z++) {
+						for (int y = 0; y < 16; y++) {
+							int id = data.readUnsignedByte();
+
+							if (id == SKULL_BLOCK_ID) {
+								skulls.put(StuffUtils.toLong(chunkXStart + x, (idx * 16) + y, chunkZStart + z), -1);
+							}
+						}
+					}
+				}
+				for (int x = 0; x < 16; x++) {
+					for (int z = 0; z < 16; z++) {
+						for (int y = 0; y < 16; y += 2) {
+							long position = StuffUtils.toLong(chunkXStart + x, (idx * 16) + y, chunkZStart + z);
+							long positionAbove = StuffUtils.toLong(chunkXStart + x, (idx * 16) + y + 1, chunkZStart + z);
+							int state = data.readUnsignedByte();
+
+							int dataValueAbove = state >> 4;
+							int dataValue = state & 0x0F;
+
+							if (skulls.containsKey(position)) {
+								skulls.put(position, dataValue);
+							}
+							if (skulls.containsKey(positionAbove)) {
+								skulls.put(positionAbove, dataValueAbove);
+							}
+						}
+					}
+				}
+			}
+
+			data.skipBytes(512); // heights
+			data.skipBytes(256); // biomes
+			data.readByte(); // borders
+			VarNumberSerializer.readSVarInt(data); // extra data
+			while (data.readableBytes() != 0) {
+				NBTTagCompoundWrapper tag = ItemStackSerializer.readTag(data, true, con.getVersion());
+
+				if (!isSkull(tag))
+					continue;
+
+				int x = tag.getIntNumber("x");
+				int y = tag.getIntNumber("y");
+				int z = tag.getIntNumber("z");
+
+				Position position = new Position(x, y, z);
+
+				// Is there's any possibility of an skull being on the chunk nbt tags but not really in the world? idk
+				// So that's why getOrDefault is used
+				handleSkull(position, tag, skulls.getOrDefault(position.asLong(), 1));
+			}
+		}
+	}
+
+	public void sendSkullSkin(byte[] skullSkin, UUID uuid, boolean self) {
+		byte[] originalSkin = Skins.INSTANCE.getSkinFromUUID(uuid).clone();
+		System.arraycopy(skullSkin, 0, originalSkin, 0, 1024 * 4);
+
+		PESkinModel model = PESkinModel.getSkinModel(false);
+
+		PocketCon.sendPocketPacket(con, new SkinPacket(
+				self ? PocketCon.getClientUniqueId(con) : uuid,
+				model.getSkinId(),
+				model.getSkinName(),
+				"Steve",
+				originalSkin,
+				new byte[0],
+				model.getGeometryId(),
+				model.getGeometryData()
+		));
+
+	}
+
+	public void sendOriginalSkin(UUID uuid, boolean self) {
+		PESkinModel model = PESkinModel.getSkinModel(false);
```

Comment:
Technically you could get the skinmodel (isslim) from the pc skin url and properties. This should however not be done in network thread.